src/main/java/com/yahoo/sketches/concurrent/theta/ConcurrentUpdateSketch.java

```diff
@@ -0,0 +1,85 @@
+package com.yahoo.sketches.concurrent.theta;
+
+import com.yahoo.sketches.theta.Sketch;
+import com.yahoo.sketches.theta.UpdateSketch;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * @author eshcar
+ */
+public class ConcurrentUpdateSketch extends UpdateSketchComposition {
+
+  private static int PARALLELISM_LEVEL = 3;
+  protected static ExecutorService PROPAGATION_EXECUTOR_SERVICE = Executors.newWorkStealingPool
+      (PARALLELISM_LEVEL);
+
+  private volatile long theta_;
+  private volatile double estimation_;
+  // A flag to coordinate between several propagation threads
+  private AtomicBoolean propagationInProgress_;
+
+  // package visibility constructors - to be created only by factory
+  ConcurrentUpdateSketch(final UpdateSketch delegattee) {
+    super(delegattee);
+    theta_ = getThetaLong();
+    estimation_ = 0;
+    propagationInProgress_ = new AtomicBoolean(false);
+  }
+
+  public void propagate(final Sketch sketchIn, AtomicBoolean propagationInProgress) {
+    BackgroundThetaPropagation job = new BackgroundThetaPropagation(sketchIn,
+        propagationInProgress);
+    PROPAGATION_EXECUTOR_SERVICE.execute(job);
+  }
+
+  public double getEstimationSnapshot() {
+    return estimation_;
+  }
+
+  public long getVolatileTheta() {
+    return theta_;
+  }
+
+  private class BackgroundThetaPropagation implements Runnable {
+
+    private Sketch sketch_;
+    private AtomicBoolean localThreadPropagationFlag_;
+
+    public BackgroundThetaPropagation(Sketch sketch, AtomicBoolean localThreadPropagationFlag) {
+      this.sketch_ = sketch;
+      this.localThreadPropagationFlag_ = localThreadPropagationFlag;
+    }
+
+    @Override public void run() {
+      assert getVolatileTheta() <= sketch_.getThetaLong();
+
+      while(!propagationInProgress_.compareAndSet(false,true)) {
+        //busy wait until can propagate
```

Comment:
Dangerous, potential for livelock.  Can give up / throw exception after so many failures?